plugin/trino-pinot/src/test/java/io/trino/plugin/pinot/BasePinotIntegrationConnectorSmokeTest.java

```diff
@@ -660,6 +662,28 @@ private void createAndPopulateNationAndRegionData(TestingKafka kafka, TestingPin
         kafka.sendMessages(nationRowsBuilder.build().stream(), schemaRegistryAwareProducer(kafka));
         pinot.createSchema(getClass().getClassLoader().getResourceAsStream("nation_schema.json"), nationTableName);
         pinot.addRealTimeTable(getClass().getClassLoader().getResourceAsStream("nation_realtimeSpec.json"), nationTableName);
+
+        // Ensure two tables are available in Pinot with expected number of rows.
+        LOGGER.info("Waiting for up to 60 secs for Pinot to load required data set.");
+        for (int i = 0; i < 60; i++) {
+            try {
+                Long regionRowsCountFromPinot =
+                        (Long) queryRunner.execute("SELECT count(*) FROM " + regionTableName).getMaterializedRows().get(0).getField(0);
+                LOGGER.info("Table [%s] has rows from Pinot: %s, from TPCH: %d", regionTableName, regionRowsCountFromPinot, regionRows.getMaterializedRows().size());
+                Long nationRowsCountFromPinot =
+                        (Long) queryRunner.execute("SELECT count(*) FROM " + nationTableName).getMaterializedRows().get(0).getField(0);
+                LOGGER.info("Table [%s] has rows from Pinot: %s, from TPCH: %d", nationTableName, nationRowsCountFromPinot, nationRows.getMaterializedRows().size());
+                if (regionRowsCountFromPinot == regionRows.getRowCount() &&
+                        nationRowsCountFromPinot == nationRows.getRowCount()) {
+                    LOGGER.info("Table [ %s ] and [ %s ] are ready to query in Pinot", regionTableName, nationTableName);
+                    break;
+                }
+            }
+            catch (Exception e) {
+                // ignore
+                Thread.sleep(1_000L);
+            }
+        }
```

Comment:
You may want to use `assertQueryEventually`.

Reply:
This is still inside the createQueryRunner, so `assertQueryEventually` will always fail with:
```
java.lang.AssertionError: Execution of 'actual' query failed: SELECT COUNT(*) FROM region

	at org.testng.Assert.fail(Assert.java:83)
	at io.trino.testing.QueryAssertions.assertQuery(QueryAssertions.java:150)
	at io.trino.testing.QueryAssertions.lambda$assertQueryEventually$0(QueryAssertions.java:218)
	at io.trino.testing.assertions.Assert.assertEventually(Assert.java:77)
	at io.trino.testing.assertions.Assert.assertEventually(Assert.java:68)
	at io.trino.testing.QueryAssertions.assertQueryEventually(QueryAssertions.java:218)
	at io.trino.testing.AbstractTestQueryFramework.assertQueryEventually(AbstractTestQueryFramework.java:273)
	at io.trino.plugin.pinot.BasePinotIntegrationConnectorSmokeTest.createAndPopulateNationAndRegionData(BasePinotIntegrationConnectorSmokeTest.java:669)
	at io.trino.plugin.pinot.BasePinotIntegrationConnectorSmokeTest.createQueryRunner(BasePinotIntegrationConnectorSmokeTest.java:176)
	at io.trino.testing.AbstractTestQueryFramework.init(AbstractTestQueryFramework.java:95)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:104)
	at org.testng.internal.Invoker.invokeConfigurationMethod(Invoker.java:515)
	at org.testng.internal.Invoker.invokeConfigurations(Invoker.java:217)
	at org.testng.internal.Invoker.invokeConfigurations(Invoker.java:144)
	at org.testng.internal.TestMethodWorker.invokeBeforeClassMethods(TestMethodWorker.java:169)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:108)
	at org.testng.TestRunner.privateRun(TestRunner.java:756)
	at org.testng.TestRunner.run(TestRunner.java:610)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:387)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:382)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:340)
	at org.testng.SuiteRunner.run(SuiteRunner.java:289)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:86)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1293)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1218)
	at org.testng.TestNG.runSuites(TestNG.java:1133)
	at org.testng.TestNG.run(TestNG.java:1104)
	at com.intellij.rt.testng.IDEARemoteTestNG.run(IDEARemoteTestNG.java:66)
	at com.intellij.rt.testng.RemoteTestNGStarter.main(RemoteTestNGStarter.java:109)
Caused by: java.lang.NullPointerException: Cannot invoke "io.trino.testing.QueryRunner.execute(io.trino.Session, String)" because "actualQueryRunner" is null
	at io.trino.testing.QueryAssertions.assertQuery(QueryAssertions.java:147)
	... 32 more
```

Reply:
I will override the init() in this class. and add the check in it.